server/controllers/contentLocation.ts

```diff
@@ -0,0 +1,57 @@
+import { frameworkAPI } from "@project-sunbird/ext-framework-server/api";
+import { Manifest } from "@project-sunbird/ext-framework-server/models";
+import { logger } from "@project-sunbird/logger";
+import { ClassLogger } from "@project-sunbird/logger/decorator";
+import { containerAPI } from "OpenRAP/dist/api";
+import * as path from "path";
+import Response from "./../utils/response";
+
+/*@ClassLogger({
+  logLevel: "debug",
+  logTime: true,
+
+})*/
+export default class ContentLocation {
+  private fileSDK;
+  private settingSDK;
+  constructor(manifestId) {
+    this.fileSDK = containerAPI.getFileSDKInstance(manifestId);
+    this.settingSDK = containerAPI.getSettingSDKInstance(manifestId);
+  }
+  public async changeContentLocation(contentPath: string) {
+
+    try {
+      const response: any = await this.settingSDK.get(`content_storage_location`);
+      response.location.push(contentPath);
+      const contentLocation = { location: response.location };
+      const status = await this.settingSDK.put(`content_storage_location`, contentLocation);
+
+      if (status) {
+        setTimeout(() => {
```

Comment:
Why setTimeout is required and we need to add the some more routes for content plugin for offline content check server.ts file for same